client/Components/CodeEditor/CodeEditor.js

```diff
@@ -0,0 +1,108 @@
+import React, { Component } from 'react'
+import axios from 'axios'
+import Select from '@material-ui/core/Select'
+import OutlinedInput from '@material-ui/core/OutlinedInput'
+import MenuItem from '@material-ui/core/MenuItem'
+import configEditor from './EditorConfigFunction'
+
+class AceEditor extends Component {
+  constructor() {
+    super()
+    this.state = {
+      code: '',
+      codeResponse: '',
+      codeError: '',
+      editor: {},
+      editorTheme: 'monokai'
+    }
+    this.configEditorBinded = configEditor.bind(this)
+  }
+
+  componentDidMount() {
+    this.setState({ editor: ace.edit('ace') }, function() {
+      const { editor } = this.state
+      const editorSession = editor.getSession()
+      this.configEditorBinded(editor, editorSession, this.state.editorTheme)
+    })
+  }
+
+  runCode = () => {
```

Comment:
Do we want to make this a thunk?

Reply:
Yep all these methods will be thunks. Same deal here with this not being fully-formed